repositories/repositories.bzl

```diff
@@ -29,71 +29,54 @@ def ros2_repositories():
         http_archive,
         name = "ros2_rcutils",
         build_file = "@com_github_mvukov_rules_ros2//repositories:rcutils.BUILD.bazel",
-        sha256 = "ce4e0148c4d759ce93fda96d36e0009ef2f2974b3ca0a67e022ce8cc078cd43f",
-        strip_prefix = "rcutils-1.1.3",
-        urls = ["https://github.com/ros2/rcutils/archive/1.1.3.tar.gz"],
+        sha256 = "e69ed32f189ed2078b22b49408f6a12b7f78d83ce5d56bdd1b6cea357ccd7e6e",
+        strip_prefix = "rcutils-5.1.1",
+        urls = ["https://github.com/ros2/rcutils/archive/5.1.1.tar.gz"],
     )
 
-    # maybe(
-    #     native.new_local_repository,
-    #     name = "ros2_rosidl",
-    #     build_file = "@com_github_mvukov_rules_ros2//repositories:rosidl.BUILD.bazel",
-    #     path = "../ros2_foxy/src/ros2/rosidl",
-    # )
-
-    # foxy branch.
     maybe(
-        new_git_repository,
+        http_archive,
         name = "ros2_rosidl",
         build_file = "@com_github_mvukov_rules_ros2//repositories:rosidl.BUILD.bazel",
-        remote = "https://github.com/mvukov/rosidl.git",
-        commit = "611c16181cfa832d2433e77c612bd75e303cca4e",
+        sha256 = "f431c394d28d926354c271e48b7d45667363309ae63c3c1bcb6275695fbc50b8",
+        strip_prefix = "rosidl-3.1.3",
+        patches = ["@com_github_mvukov_rules_ros2//repositories/patches:rosidl-3.1.3.patch"],
+        patch_args = ["-p1"],
+        urls = ["https://github.com/ros2/rosidl/archive/3.1.3.tar.gz"],
```

Comment:
It might be good to have mirrors for these kinds of things, but I guess the ROS project is unlikely to delete these